src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefLocalTests.cs

```diff
@@ -3069,6 +3069,243 @@ .maxstack  4
 }");
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71369")]
+        public void AssignmentValue_RefField()
+        {
+            var source = """
+                int n = 0;
+                System.Console.Write(new S(ref n).GetI());
+                System.Console.Write(n);
+
+                ref struct S
+                {
+                    ref int i;
+                    public S(ref int n) => i = ref n;
+                    public int GetI() => i = 2;
+                }
+                """;
+            var verifier = CompileAndVerify(source, expectedOutput: ExecutionConditionUtil.IsMonoOrCoreClr ? "22" : null,
+                verify: Verification.FailsPEVerify, targetFramework: TargetFramework.Net70);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("S.GetI", """
+                {
+                  // Code size       12 (0xc)
+                  .maxstack  3
+                  .locals init (int V_0)
+                  IL_0000:  ldarg.0
+                  IL_0001:  ldfld      "ref int S.i"
+                  IL_0006:  ldc.i4.2
+                  IL_0007:  dup
+                  IL_0008:  stloc.0
+                  IL_0009:  stind.i4
+                  IL_000a:  ldloc.0
+                  IL_000b:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71369")]
+        public void AssignmentValue_RefField_Temp()
+        {
+            var source = """
+                int n = 0;
+                System.Console.Write(new S(ref n).GetI());
+                System.Console.Write(n);
+
+                ref struct S
+                {
+                    ref int i;
+                    public S(ref int n) => i = ref n;
+                    public int GetI()
+                    {
+                        int x = i = 2;
```

Comment:
Is there a test where the RHS of the assignment has a side effect to make sure we handle the spill properly? Don't need a full suite here, just want to make sure there is at least one test that does this.